uiowa/uiowa
# Initial framework for engineering migrate module.

Relates to #2893.

Initial work for an engineering migration module for use with engineering articles.





# How to test



Need a local copy of www.engineering.uiowa.edu in Dev Desktop (on uiowa703, listed with the www).
Leave the DD Apache/MySQL stack running.

`git checkout engineering-migrate` and `git pull`
`blt ds --site=engineering.uiowa.edu`
`drush @engineering.local en engineering_migrate`
`drush @engineering.local mim d7_engineering_article`

It will most likely terminate. 
 - If it terminates with a ` [warning] Memory usage is now 442.78 MB (86% of limit 512 MB), not enough reclaimed, starting new batch` you can simply run the `mim` command above again. 
 - If it terminates with a `Drush command terminated abnormally`, you will have to run: 
     - `drush @engineering.local mst d7_engineering_article` to stop the migration, then 
     - `drush @engineering.local mrs d7_engineering_article` to reset the migration to Idle, 
     - and then run the `mim` command again.
   
`drush @engineering.local mim d7_engineering_article_redirect`

Check that migration completes successfully. Currently there is a discrepancy in row counts--a 'complete' migration should import 1139 articles.

Spot check articles and see that nothing looks out of place, that articles from the existing site which had featured images have featured images post-migration.

For inline images, can check the database for any body fields containing a `` token.

Spot check for any other gotchas that you can think to check for that may not have migrated over correctly.

With the redirect migrate, can start at https://www.engineering.uiowa.edu/news, pick and choose articles, and use the same aliases on the prod site.
